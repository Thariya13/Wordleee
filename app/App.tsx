/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 */

import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {StatusBar} from 'react-native';
import {Provider} from 'react-redux';

import HomeStackNavigator from './navigation/home/HomeNavigation';
import {store} from './store/Store';

import {LogBox} from 'react-native';
LogBox.ignoreLogs(['Warning: ...']); // Ignore log notification by message
LogBox.ignoreAllLogs(); //Ignore all log notifications

function App(): JSX.Element {
  return (
    <Provider store={store}>
      <NavigationContainer>
        <StatusBar barStyle={'dark-content'} backgroundColor={'white'} />
        <HomeStackNavigator />
      </NavigationContainer>
    </Provider>
  );
}

export default App;
