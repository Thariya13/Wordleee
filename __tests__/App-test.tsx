/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../app/App';
import HeaderTitle from '../app/shared/components/headerTitle/HeaderTitle.component';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import WordleeeHeader from '../app/shared/components/wordleeeHeader/WordleeeHeader.component';
import {Text} from 'react-native';
import WordleeeButton from '../app/shared/components/wordleeeButton/WordleeeButton.component';
import TouchableOpacityDebounce from '../app/shared/components/touchableOpacityDebounce/TouchableOpacityDebounce.component';

it('renders correctly', () => {
  renderer.create(<App />);
});

test('Header title renders correctly', () => {
  const tree = renderer.create(<HeaderTitle title="Home Screen" />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Wordlee Header renders correctly', () => {
  const tree = renderer
    .create(
      <WordleeeHeader>
        <Text>Hello World</Text>
      </WordleeeHeader>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Wordlee Button renders correctly', () => {
  const tree = renderer
    .create(
      <WordleeeButton
        title="Hello"
        activeButtonStyle={{backgroundColor: 'red'}}
        handleOnPress={() => console.log('Hello World')}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Touchable Opacity Debounce renders correctly', () => {
  const tree = renderer
    .create(
      <TouchableOpacityDebounce onPress={() => console.log('Hello World')}>
        <Text>Hello World</Text>
      </TouchableOpacityDebounce>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
