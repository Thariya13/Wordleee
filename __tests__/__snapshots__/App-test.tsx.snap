// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header title renders correctly 1`] = `
<Text
  style={
    {
      "fontSize": 10,
    }
  }
>
  Home Screen
</Text>
`;

exports[`Touchable Opacity Debounce renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <Text>
    Hello World
  </Text>
</View>
`;

exports[`Wordlee Button renders correctly 1`] = `
<View
  style={
    {
      "paddingHorizontal": 20,
      "paddingVertical": 10,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "aspectRatio": 5.454545454545454,
        "backgroundColor": "red",
        "borderColor": "#5544EE",
        "borderRadius": 5,
        "borderWidth": 2,
        "justifyContent": "center",
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "color": undefined,
          "fontSize": 20,
        }
      }
    >
      Hello
    </Text>
  </View>
</View>
`;

exports[`Wordlee Header renders correctly 1`] = `
<View
  style={
    {
      "flex": 1,
      "paddingTop": 70,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
      }
    }
  >
    <Text
      style={
        [
          {
            "fontSize": 60,
            "fontWeight": "300",
            "letterSpacing": -0.5,
          },
          {
            "color": "#000000",
          },
          {
            "color": "#5544EE",
            "fontFamily": "Roboto-Bold",
          },
        ]
      }
    >
      Wordleee
    </Text>
  </View>
  <Text>
    Hello World
  </Text>
</View>
`;
